### DISCUSSION TITLE
Access URL parameters using getStaticProps?

### DISCUSSION THREAD

**Discussion by johanscreencloud** [vercel/next.js#25002] [Help] [answered]
Hi,

I'm building a blog with several pages which will need to have the URL '/blog' and '/blog?page=1', '/blog?page=2' etc. to keep SEO for a previous site.

Currently I'm using getServerSideProps to get the ?page=1 param and render the correct page of results for the user. However, many have told me to use getServerSideProps as little as possible. But with getStaticProps I can't get the { query } parameter obviously as it is executed at build-time.

Is there a way of using getStaticProps that correctly will render a page if a user uses a URL with a parameter like
 '/blog?page=1'?

Thanks,

**Comment by bobarros** [comment, marked as answer]
Can't you handle the logic inside the component/page and get queries? With:

```jsx
import { useRouter } from 'next/router';
```

**Reply by johanscreencloud** [reply]
Great tip - I'll just load the initial (page 1) set of blog posts using getStaticProps but if anyone uses a ?page=2 query, it fetches them then in realtime using the api instead. Thanks.
